### DISCUSSION TITLE
Video Stream example?

### DISCUSSION THREAD

**Discussion by numairawan** [jimmywarting/StreamSaver.js#282] [Q&A]
Hi, can i get video stream example?

**Comment by jimmywarting** [comment]
You mean from webcam?

**Comment by numairawan** [comment]
NO, from the mp4 file llink. But i want to play in chunks so need to download the chunks and save in disk then play from there. Possible?

**Reply by jimmywarting** [reply]
if you are downloading something from the server then i REALLY recommend you to solve this on the server side instead of doing any browser js solution 

Add the following headers to the response:

- **content-type: application/octet-stream**
  - will make it so browser will be incompitable of rendering the media
- **content-length: {size in bytes}**
  - useful for knowing how large the size is and knowing the speed/est and being able to make range request to failed downloads 
- **content-disposition: attachment; filename="filename.mp4"**
  - give browser a hint of what the filename should be
- **accept-ranges: bytes**
  - Will make it possible for browser to pause/resume and retry failed downloads 

doing this on the server is WAY better than doing it on the client side as it will be able to download it in the background thread also so the page dose not need to be open all the time

**Comment by numairawan** [comment]
So i cant do this with streamsaver right?

**Reply by jimmywarting** [reply]
you can if you want to

**Comment by numairawan** [comment]
I am doing with blob but there are some issue, but can i use URL.createObjectURL with streamsaver to generate link of stored data in disk?

**Reply by jimmywarting** [reply]
if you already have a blob, then you can just as well use `a[download]`
```js
a = document.createElement('a')
a.download = 'movie.mp4'
a.href = URL.createObjectURL(blob)
a.click()
```
